File: src/main/java/main/utils/json/autoplay/AutoPlayConfig.java
package main.utils.json.autoplay;

import main.utils.json.AbstractGuildConfig;
import net.dv8tion.jda.api.entities.Guild;
import org.bson.Document;
import org.json.JSONException;
import org.json.JSONObject;

public class AutoPlayConfig extends AbstractGuildConfig {
    private final Guild guild;
    private final long gid;

    public AutoPlayConfig(Guild guild) {
        super(guild);
        this.guild = guild;
        this.gid = guild.getIdLong();
    }

    public boolean getStatus() {
        try {
            return getGuildObject().getBoolean(Field.AUTOPLAY.name().toLowerCase());
        } catch (JSONException e) {
            update();
            return getGuildObject().getBoolean(Field.AUTOPLAY.name().toLowerCase());
        }
    }

    public void setStatus(boolean status) {
        JSONObject guildObject = getGuildObject();
        guildObject.put(Field.AUTOPLAY.name().toLowerCase(), status);
        getCache().updateGuild(guildObject, gid);
    }

    @Override
    public void update() {
        JSONObject guildObject = getGuildObject();

        if (!guildObject.has(Field.AUTOPLAY.name().toLowerCase()))
            guildObject.put(Field.AUTOPLAY.name().toLowerCase(), false);
        getCache().updateCache(String.valueOf(gid), Document.parse(guildObject.toString()));
    }

    public enum Field {
        AUTOPLAY
    }
}

File: src/main/kotlin/main/utils/json/autoplay/AutoPlayConfig.kt
package main.utils.json.autoplay

import main.utils.json.AbstractGuildConfig
import net.dv8tion.jda.api.entities.Guild
import org.bson.Document
import org.json.JSONException
import java.util.*

class AutoPlayConfig(private val guild: Guild) : AbstractGuildConfig(guild) {

    fun getStatus(): Boolean {
        return getGuildModel().autoplay ?: false
    }

    fun setStatus(status: Boolean) {
        cache.updateGuild(guild.id) {
            autoplay = status
        }
    }

    override fun update() {
        val guildObject = getGuildModel().toJsonObject()
        if (!guildObject.has(Field.AUTOPLAY.name.lowercase(Locale.getDefault())))
            guildObject.put(Field.AUTOPLAY.name.lowercase(), false)
        cache.updateCache(guild.idLong.toString(), Document.parse(guildObject.toString()))
    }


    enum class Field {
        AUTOPLAY
    }

}